platforms/quarkus/runtime/src/main/java/io/hawt/quarkus/filters/HawtioQuarkusAuthenticationFilter.java

```diff
@@ -31,33 +33,79 @@ public class HawtioQuarkusAuthenticationFilter extends AuthenticationFilter {
 
     private static final Logger LOG = LoggerFactory.getLogger(HawtioQuarkusAuthenticationFilter.class);
 
+    private InstanceHandle<HawtioQuarkusAuthenticator> handle;
     private HawtioQuarkusAuthenticator authenticator;
 
     @Override
     public void init(FilterConfig filterConfig) throws ServletException {
-        authenticator = Arc.container().instance(HawtioQuarkusAuthenticator.class).get();
+        handle = Arc.container().instance(HawtioQuarkusAuthenticator.class);
+        authenticator = handle.get();
```

Comment:
I think then this `handle` should be renamed to `authenticator` and the original raw `authenticator` should be removed in favour of always doing `authenticator.get()` when needed.

Reply:
Now I don't think it's a good idea. Your original one is better. Let me close it.

Reply:
You forgot we're not in OSGi ;)